docs/dev/plugin-quickstart.rst

```diff
@@ -0,0 +1,914 @@
+    model components in the :ref:`Model class and components <plugin_components>` section.
+  - **Methods and workflows**: Most of the heavy work in HydroMT is done by Methods and workflows, indicated by the gear wheels in the schematic of
+    HydroMT architecture below. Methods provide the low-level functionality such as GIS rasterization, reprojection, or zonal statistics. Workflows combine
+    several methods to transform data to a model layer or postprocess model results.
+
+    In the :ref:`data processing section <plugin_workflows>`, we will see
+    some examples and best practices of how to process and add data to your model.
+  - **Command line and Python interface**: The CLI is a high-level interface to HydroMT. It is used to run HydroMT methods such as build, update or clip.
+    While most common functionalities can be called through the CLI, the Python interface offers more flexibility for advanced users. It allows you to e.g.
+    interact directly with a model component Model API and apply the many methods and workflows available.
+
+    We saw in the :ref:`section above <plugin_create>`
+    how to properly instantiate your plugin class and register it to HydroMT entrypoints.
+
+.. figure:: ../_static/Architecture_model_data_input.png
+
+  Schematic of HydroMT architecture
+
+.. _plugin_components:
+
+Model class and components
+--------------------------
+In this section, we will detail the bottom part of the schematic ie how to initialise your HydroMT Model class for your plugin, the main properties and how to
+work with its model components.
+
+Initialisation
+^^^^^^^^^^^^^^
+In the :ref:`create plugin section <plugin_create>`, we already saw which HydroMT ``Model`` class to choose for your plugin. Here we will focus on additionnal
+properties and the initiliasation of your new Model subclass.
+
+To fully initiliase your new subclass (eg *MyModelModel*), you need to initialise a couple of high level properties and in some cases, you may wish to modify
+the default initialisation function of you parent class (the HydroMT core class you choose, ``Model`` or ``GridModel`` etc.).
+
+.. TIP::
+
+  In python, a child class (eg *MyModelModel(Model)*), inherits all methods and properties of the parent class (``Model``). If you wish to overwrite completely one
+  of the parent method (eg ``setup_region``), you can just redefine the function in the child class using the name (``setup_region``). You can also decide to partly
+  re-use the parent function and for example add some extra steps, change the default values or the docstring of the function. For this, you can redefine the
+  function in the child class, and within the redefined function, call the parent one use *super()* python class attribute (*super().setup_region*). You will see
+  an example below for the initialisation function ``__init__`` .
+
+Below are the properties and initialisation function, to set for your new plugin *MyModelModel* class:
+
+.. code-block:: python
```

Comment:
I wonder if we should have this here as it will require keeping it in sync with the cookiecutter. Perhaps a notebook using the cookiecutter would be more suitable for this level docs? In such notebook you can also demonstrate the inheritance of properties, available methods, etc.

Reply:
I agree with this sentiment but I have no idea how good the compatibility between cookiecutter and notebooks are, but I don't imagine it's great and honestly I'm not very keen on the idea. I'm not sure having it included here adds that much so I just removed it. I think people can just use the template for this if they are curious.